E3db/Classes/Document.swift

```
@@ -0,0 +1,214 @@
+    public func encrypt(type: String, data: RecordData, eakInfo: EAKInfo, plain: PlainMeta? = nil) throws -> EncryptedDocument {
+        let clientId  = config.clientId
+        let meta      = ClientMeta(writerId: clientId, userId: clientId, type: type, plain: plain)
+        let recInfo   = DocInfo(meta: meta, data: data)
+        let localAk   = try getLocalAk(clientId: clientId, recordType: type, eakInfo: eakInfo)
+        let signed    = try sign(document: recInfo)
+        let encrypted = try Crypto.encrypt(recordData: recInfo.data, ak: localAk)
+        return EncryptedDocument(clientMeta: meta, encryptedData: encrypted, recordSignature: signed.signature)
+    }
+
+    /// Create a document to hold the original plaintext data from the given encrypted format.
+    /// The resulting document also holds info related to the author, type of data, and any additional
+    /// metadata kept in cleartext. The input document is also verified for authenticity with the given
+    /// public signing key, and throws an error if verification fails.
+    ///
+    /// - Parameters:
+    ///   - encryptedDoc: Data type to hold encrypted data and related info
+    ///   - eakInfo: The encrypted access key information used for the decryption operation
+    /// - Returns: Data type to hold the unencrypted data and related info
+    /// - Throws: `E3dbError.cryptoError` if the decrypt operation fails, or if the document fails verification
+    public func decrypt(encryptedDoc: EncryptedDocument, eakInfo: EAKInfo) throws -> DecryptedDocument {
+        let meta      = encryptedDoc.clientMeta
+        let localAk   = try getLocalAk(clientId: eakInfo.authorizerId, recordType: meta.type, eakInfo: eakInfo)
+        let decrypted = try Crypto.decrypt(cipherData: encryptedDoc.encryptedData, ak: localAk)
+
+        // attempt to verify if a signing key is present in the EAKInfo instance
+        if let sigKey = eakInfo.signerSigningKey?.ed25519 {
```

I can't find any record of it, but I thought we had settled on reporting verification errors via a handler so the client could deal with them. In any case I think silent failure is not good.